main.tf

```diff
@@ -35,22 +37,29 @@ resource "aws_security_group" "this" {
     cidr_blocks = ["0.0.0.0/0"]
   }
 
-  tags = var.tags
+  tags = merge(var.tags, {Name = "vpc all interface endpoints", env = terraform.workspace})
+}
+
+resource "aws_security_group" "all" {
+  count = var.create_vpc_endpoints ? length(local.vpc_endpoint_interfaces) : 0
+
+  description = "VPC Interface ${local.vpc_endpoint_interfaces[count.index]} Endpoint"
+  vpc_id      = local.vpc_id
+
+  tags = merge(var.tags, {Name = "vpc ${local.vpc_endpoint_interfaces[count.index]} interface endpoint", env = terraform.workspace})
 }
 
 resource "aws_vpc_endpoint" "interface_services" {
-  count = var.create_vpc_endpoints ? length(var.vpc_endpoint_interfaces) : 0
+  count = var.create_vpc_endpoints ? length(local.vpc_endpoint_interfaces) : 0
 
   vpc_id            = local.vpc_id
-  service_name      = var.vpc_endpoint_interfaces[count.index]
+  service_name      = length(regexall(local.vpc_endpoint_interfaces[count.index], data.aws_region.current.name)) > 0 ? local.vpc_endpoint_interfaces[count.index] : "com.amazonaws.${data.aws_region.current.name}.${var.vpc_endpoint_interfaces_service_short_names[count.index]}"
```

Comment:
What is the value of doing this in the module, instead of constructing the endpoint names in the user's config?

If there is real value in doing it in the module, then I think it makes sense to keep the interface simple and change `var.vpc_endpoint_interfaces` to require short names. It is backwards incompatible, but that just means a major version bump. No big deal.

Reply:
For me, the value is ease of use out of the box. I assume, as an example, most users will initially just set:
```
  vpc_endpoint_interfaces = ["com.amazonaws.us-east-1.apigateway"]
```

For one, it was not obvious to me that the endpoint needed to be declared TLD first, followed by the subdomain, as the Endpoint is listed as `apigateway.us-east-1.amazonaws.com` here: https://docs.aws.amazon.com/general/latest/gr/rande.html .

Second, when deploying into a new region, one does not need to think of writing this code instead:
```
  vpc_endpoint_interfaces = ["com.amazonaws.${data.aws_region.current.name}.apigateway"]
```

If one is not using region name for anything else, that is also an additional data source declaration. It's not a groundbreaking change, but I think makes the module just that much easier to use.

Reply:
I'd have some concerns that the service name is not the same in all partitions, also. E.g. commercial (aws) vs govcloud (aws-us-gov) vs china (aws-cn), etc... AWS hasn't always been consistent about endpoint naming, so it is hard in a module to cover/anticipate all a user's use cases.

It is frustrating the service names are different from the endpoint urls, and that the service names do not seem to be documented. Though they are discoverable using `aws ec2 describe-vpc-endpoint-services`.

Reply:
And then there are custom interface endpoints, aka PrivateLink, which I presume have their own service name syntax.

Reply:
oh, good call with the `describe-vpc-endpoint-services`. Looks like there's a terraform data source, so that should solve your concerns?

I also renamed the var to better reflect the input expected from the user. I'm assuming since this is a breaking change that's fine, but I can keep the original name if you prefer.

Reply:
That'll work. Still not sure about custom interface endpoints, but have nothing to test with at the moment anyway, so can worry about that as a future use case.